examples/nlp/gpt/train_gpt_sft.py

```diff
@@ -119,15 +126,34 @@ def main(cfg) -> None:
     with open_dict(cfg):
         cfg.model.precision = cfg.trainer.precision
 
-    ptl_model = load_from_nemo(
+    assert cfg.model.peft.peft_scheme in ["lora", "none"], "Only support LoRA or Full finetuning"
+    if cfg.model.peft.peft_scheme == "lora":
+        assert (
+            cfg.model.optim.name != "distributed_fused_adam"
+        ), "LoRA doesn't support distributed_fused_adam, please use fused_adam"
+
+    ptl_model, updated_cfg = load_from_nemo(
         GPTSFTModel,
         cfg,
         trainer,
         strict=True,
         modify_config_fn=_modify_config,
         restore_path=cfg.model.restore_from_path,
+        return_updated_cfg=True,
     )
 
+    peft_cfg_cls = PEFT_CONFIG_MAP[cfg.model.peft.peft_scheme]
+    if cfg.model.peft.restore_from_path is not None:
+        # initialize peft weights from a checkpoint instead of randomly
+        # This is not the same as resume training because optimizer states are not restored.
+        logging.info("PEFT Weights will be loaded from", cfg.model.peft.restore_from_path)
+        ptl_model.load_adapters(cfg.model.peft.restore_from_path, peft_cfg_cls(updated_cfg))
+    elif peft_cfg_cls is not None:
+        logging.info("Adding adapter weights to the model for PEFT")
+        ptl_model.add_adapter(peft_cfg_cls(updated_cfg))
+    else:
+        logging.info(f"Running full finetuning since no peft scheme is given.\n{ptl_model.summarize()}")
```

Comment:
i believe we also want to use PEFT for other algorithms as well, so I think we should create a `initialize_peft` function in `train_script_utils` to handle all this logic related to PEFT(such as error checking, and loading the adaptors)

Reply:
Good point, 09ad8c61852debbbd2e713396f4b99f1441b52f5